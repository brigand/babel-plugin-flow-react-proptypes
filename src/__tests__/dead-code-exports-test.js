const babel = require('babel-core');
const { minify } = require('uglify-js');
const content = `
var React = require('react');

export type MyType = string | number;

`;

const opts = {
  babelrc: false,
  presets: ['es2015', 'stage-1', 'react'],
  plugins: ['syntax-flow', [require('../'), { deadCode: '__PROD__' }]],
};

it('dead-code-exports', () => {
  const res = babel.transform(content, opts).code;
  expect(res).toMatch(/__PROD__/);
  expect(res).toMatchSnapshot();
});

it('dead-code-exports uglify', () => {
  const res = babel.transform(content, opts).code
    .replace(/__PROD__/g, 'true');
  const { code: min } = minify(res, { toplevel: true });
  expect(min).not.toMatch(/prop-types/);
  expect(min).toMatchSnapshot();
});
