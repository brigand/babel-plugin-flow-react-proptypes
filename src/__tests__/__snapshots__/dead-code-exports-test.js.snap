// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dead-code-exports 1`] = `
"var bpfrpt_proptype_MyType = __PROD__ ? null : require('prop-types').oneOfType([require('prop-types').string, require('prop-types').number, function () {
  return (typeof require('./dep.js').bpfrpt_proptype_DepType === 'function' ? require('./dep.js').bpfrpt_proptype_DepType : require('prop-types').shape(require('./dep.js').bpfrpt_proptype_DepType)).apply(this, arguments);
}]);
if (!__PROD__ && typeof exports !== 'undefined') Object.defineProperty(exports, 'bpfrpt_proptype_MyType', {
  value: bpfrpt_proptype_MyType,
  configurable: true
});"
`;

exports[`dead-code-exports uglify 1`] = `""`;

exports[`dead-code-exports uglify with esm 1`] = `"var p=null;import{bpfrpt_proptype_DepType as r}from\\"./dep.js\\";import o from\\"prop-types\\";export{p as bpfrpt_proptype_MyType};"`;

exports[`dead-code-exports with esm 1`] = `
"var bpfrpt_proptype_MyType = __PROD__ ? null : PropTypes.oneOfType([PropTypes.string, PropTypes.number, function () {
  return (typeof bpfrpt_proptype_DepType === 'function' ? bpfrpt_proptype_DepType : PropTypes.shape(bpfrpt_proptype_DepType)).apply(this, arguments);
}]);
import { bpfrpt_proptype_DepType } from './dep.js';
import PropTypes from 'prop-types';
export { bpfrpt_proptype_MyType };"
`;
